### PR TITLE
updated oauth flow

### PR DESCRIPTION
**Pull request by morgnism**
Updated OAuth flow to verify in a tab. The client should use polling to fetch the access code as a temporary stop-gap for now until we figure out how to redirect from the server to the app.

TODO:
* Move RedirectCoponent logic to a guard (AuthGuard)
* Have @dustinsgoodman change the redirect uri in Github OAuth to point at `https://<domain_host>`.

### CODE REVIEW

File: angular-apollo-tailwind/src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -76,6 +71,7 @@ export class AuthService {
    * @memberof AuthService
    */
   isAuthenticated(): boolean {
+    console.log('isAuth: ', !!this.tokenService.getToken());
```
Comment (ktrz) [inline angular-apollo-tailwind/src/app/auth/auth.service.ts:74]:
```suggestion
```

File: angular-apollo-tailwind/src/environments/environment.prod.ts [hunk 2]
```diff
@@ -2,4 +2,5 @@ export const environment = {
   production: true,
   apiUrl: 'https://fhtfslckqb.execute-api.us-east-1.amazonaws.com/api',
   graphApiUrl: 'https://api.github.com/graphql',
+  redirectUrl: 'https://localhost:4200/redirect',
```
Comment (ktrz) [inline angular-apollo-tailwind/src/environments/environment.prod.ts:5]: the prod redirectUrl should probably be different

File: starter-dev-backend/lib/clear-cookies.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { ACCESS_TOKEN_COOKIE } from './constants';
+
+export default (req, res) => {
+  res
+    .status(200)
+    .clearCookie(ACCESS_TOKEN_COOKIE, {
+      sameSite: 'strict',
```
Comment (ktrz) [inline starter-dev-backend/lib/clear-cookies.js:7]:
```suggestion
      sameSite: 'lax',
```

**Review by ktrz — CHANGES_REQUESTED** [review]
added a couple of small suggestions. Please remove `starter-dev-backend/package-lock.json` file as we should only keep one lock file (`yarn.lock` in our case)

File: starter-dev-backend/serverless.yml [hunk 4]
```diff
@@ -9,6 +9,11 @@ plugins:
 custom:
   serverless-offline:
     httpPort: 4000
+  customDomain:
+    domainName: angular-apollo-tailwind.starter.dev
```
Comment (dustinsgoodman) [inline starter-dev-backend/serverless.yml:13]: This should be `api.starter.dev`

**Review by dustinsgoodman — CHANGES_REQUESTED** [review]

File: angular-apollo-tailwind/src/environments/environment.prod.ts [hunk 5]
```diff
@@ -2,5 +2,6 @@ export const environment = {
   production: true,
   apiUrl: 'https://fhtfslckqb.execute-api.us-east-1.amazonaws.com/api',
   graphApiUrl: 'https://api.github.com/graphql',
-  redirectUrl: 'https://localhost:4200/redirect',
+  redirectUrl:
+    'https://fhtfslckqb.execute-api.us-east-1.amazonaws.com/redirect',
```
Comment (dustinsgoodman) [inline angular-apollo-tailwind/src/environments/environment.prod.ts:6]: Is there any way to shift this to be from a `.env` file so it's not hardcoded?

File: starter-dev-backend/package.json [hunk 6]
```diff
@@ -13,6 +13,7 @@
     "crypto": "^1.0.1",
     "express": "^4.17.1",
     "morgan": "^1.10.0",
+    "serverless-domain-manager": "^6.0.2",
```
Comment (dustinsgoodman) [inline starter-dev-backend/package.json:16]: can we move to dev deps instead of main deps?

**Review by dustinsgoodman — APPROVED** [review]

**Review by ktrz — APPROVED** [review]

**Pull request merged**